actors/common/src/main/java/org/sunbird/common/quartz/scheduler/OnDemandShadowManager.java

```diff
@@ -0,0 +1,216 @@
+package org.sunbird.common.quartz.scheduler;
+
+import org.apache.commons.lang3.StringUtils;
+import org.quartz.*;
+import org.quartz.impl.StdSchedulerFactory;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+import static org.quartz.SimpleScheduleBuilder.simpleSchedule;
+
+public class OnDemandShadowManager {
+
+  private static final String FILE = "quartz.properties";
+  private static Scheduler scheduler = null;
+  private static org.sunbird.common.quartz.scheduler.OnDemandShadowManager OnDemandShadowManager = null;
+
+  private OnDemandShadowManager() {
+   schedule(null);
+  }
+
+  /** This method will register the quartz scheduler job. */
+  private void schedule(String jobs) {
+   ProjectLogger.log(
+           "OnDemandShadowManager:schedule: Call to start scheduler jobs - org.sunbird.common.quartz.scheduler.OnDemandShadowManager",
+           LoggerEnum.INFO.name());
+
+   try {
+    boolean isEmbedded = false;
+    Properties configProp = null;
+    String embeddVal = System.getenv(JsonKey.SUNBIRD_QUARTZ_MODE);
+    if (JsonKey.EMBEDDED.equalsIgnoreCase(embeddVal)) {
+     isEmbedded = true;
+    } else {
+     configProp = setUpClusterMode();
+    }
+    if (!isEmbedded && configProp != null) {
+     ProjectLogger.log("Quartz scheduler is running in cluster mode.", LoggerEnum.INFO.name());
+     scheduler = new StdSchedulerFactory(configProp).getScheduler();
+    } else {
+     ProjectLogger.log("Quartz scheduler is running in embedded mode.", LoggerEnum.INFO.name());
+     scheduler = new StdSchedulerFactory().getScheduler();
+    }
+
+   } catch (Exception e) {
+    ProjectLogger.log(
+            "OnDemandShadowManager:schedule: Error in starting scheduler jobs - org.sunbird.common.quartz.scheduler.OnDemandShadowManager "
+                    + e.getMessage(),
+            LoggerEnum.ERROR.name());
+   } finally {
+    registerShutDownHook();
+   }
+   ProjectLogger.log(
+           "OnDemandShadowManager:schedule: started scheduler jobs - org.sunbird.common.quartz.scheduler.OnDemandShadowManager",
+           LoggerEnum.INFO.name());
+  }
+
+  private void scheduleBulkUploadJobOnDemand(String identifier) {
```

Comment:
We can make it better , instead of writing separate method for each job run , we can use one method that will accept jobname  and jobclass as map .

Reply:
done